samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -167,6 +172,20 @@ public String getProcessorId() {
     return processorId;
   }
 
+  @Override
+  public Set<String> getProcessorNames() {
+    final Set<String> processorNames = new HashSet<>();
+    // Leader election should have happened by this time.
+    // SAMZA-1522 Add an API to check if leader election is done.
+    ZkConfig zkConfig = new ZkConfig(config);
+    ZkClient zkClient =
+        createZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
+    ZkKeyBuilder keyBuilder = new ZkKeyBuilder(getJobCoordinationZkPath(config));
+    int numProcessors = zkClient.countChildren(keyBuilder.getProcessorsPath());
+    IntStream.range(0, numProcessors).forEach(i -> processorNames.add(String.valueOf(i)));
```

Comment:
Just curious and want to understand this better.
A.  We already have processorId with which each processor joins the group. Wouldn't that processorIDList be sufficient here?
B.  If we simply count children and generate processNames based upon that, we wouldn't be able to distinguish different processor processorId groups(say processorIds: [1,2,3,4] vs processorIds:  [1,2,3,5])?